src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,90 @@
+import cn from 'classnames';
+import { Link, useSearchParams } from 'react-router-dom';
+
+const CENTURIES = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const currentQuery = searchParams.get('query') || '';
+  const currentSex = searchParams.get('sex') || '';
+  const centuries = [...searchParams.getAll('centuries')];
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
+    const newParams = new URLSearchParams();
+
+    if (inputValue) {
+      newParams.set('query', inputValue);
+    } else {
+      newParams.delete('query');
+    }
+
+    setSearchParams(newParams);
+  };
+
+  const handleSortMale = (field: string) => {
+    const param = new URLSearchParams();
+
+    if (field === 'sex') {
+      param.set('sex', 'm');
+    }
+
+    return `/people?${param.toString()}`;
+  };
+
+  const handleSortFemale = (field: string) => {
+    const param = new URLSearchParams();
+
+    if (field === 'sex') {
+      param.set('sex', 'f');
+    }
+
+    return `/people?${param.toString()}`;
+  };
+
+  const toggleCentury = (num: string) => {
+    if (centuries.includes(num)) {
+      return centuries.filter(century => century !== num);
+    }
+
+    return [...centuries, num];
+  };
+
+  const handleSortCenuties = (field: string) => {
+    const param = new URLSearchParams();
+
+    const selectedCenturies = toggleCentury(field);
+
+    selectedCenturies.forEach(century => param.append('centuries', century));
+
+    return `/people?${param.toString()}`;
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
```

Comment:
You can use a SearchLink component instead 
![image](https://github.com/mate-academy/react_people-table-advanced/assets/99403198/f3e7e38a-3b5e-40de-999b-ee987d5bd887)

Reply:
Not fixed. Feel free to ask questions in the chat if you need help 
![image](https://github.com/mate-academy/react_people-table-advanced/assets/99403198/838a6021-7802-440d-b16a-e73e802627ac)